__tests__/integrations/FacebookPixel.test.js

```diff
@@ -0,0 +1,234 @@
+import FacebookPixel from "../../src/integrations/FacebookPixel/browser";
+
+beforeAll(() => { });
+
+afterAll(() => {
+  jest.restoreAllMocks();
+});
+
+const mockAnalyticsIdentifiedUser = {
+  getUserTraits: jest.fn(() => ({
+    firstName: "rudder",
+    lastName: "stack",
+    email: "abc@rudderstack.com"
+  })),
+  getAnonymousId: jest.fn(() => "testAnonymousID"),
+  getUserId: jest.fn(() => "testUserID")
+};
+
+const mockAnalyticsAnonymousUser = {
+  getUserTraits: jest.fn(() => null),
+  getAnonymousId: jest.fn(() => "testAnonymousID"),
+  getUserId: jest.fn(() => null)
+};
+
+describe("FacebookPixel init tests", () => {
+  let facebookPixel;
+
+  test("Testing init call of Facebook Pixel with identified user", () => {
+    facebookPixel = new FacebookPixel({ pixelId: "12567839", useUpdatedMapping: true }, mockAnalyticsIdentifiedUser);
+    facebookPixel.init();
+    expect(typeof window.fbq).toBe("function");
+    expect(facebookPixel.userPayload).toStrictEqual({
+        "db": undefined,
+        "em": "abc@rudderstack.com",
+        "external_id": "d06773e1bf4b8a96a4786fbb8e3444092438ad29401769613ae9e0e3e1e08a84",
+        "fn": "rudder",
+        "ge": undefined,
+        "ln": "stack", 
+        "ph": undefined
+    });
+  });
+
+  test("Testing init call of Facebook Pixel with anonymous user", () => {
+    facebookPixel = new FacebookPixel({ pixelId: "12567839", useUpdatedMapping: true }, mockAnalyticsAnonymousUser);
+    facebookPixel.init();
+    expect(typeof window.fbq).toBe("function");
+    expect(facebookPixel.userPayload).toStrictEqual(
+        {"db": undefined, "em": undefined, "external_id": "d4e669b42293a60cc65b22830a922824e6e9c7736c6058fbbb3de780d2f4a17f", "ge": undefined, "ph": undefined}
+    );
+  });
+});
+
+describe("FacebookPixel page", () => {
+  let facebookPixel;
+  beforeEach(() => {
+    facebookPixel = new FacebookPixel({ pixelId: "12567839", useUpdatedMapping: true }, mockAnalyticsIdentifiedUser);
```

Comment:
declare the mocked analytics for page tests within test scope it self

Reply:
updated